flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -201,8 +201,11 @@ private SplitEnumerator<IcebergSourceSplit, IcebergEnumeratorState> createEnumer
       return new ContinuousIcebergEnumerator(
           enumContext, assigner, scanContext, splitPlanner, enumState);
     } else {
-      List<IcebergSourceSplit> splits = planSplitsForBatch(planningThreadName());
-      assigner.onDiscoveredSplits(splits);
+      if (enumState == null) {
```

Comment:
thanks for fixing the bug.

maybe add a comment like `Only do scan planning if nothing is restored from checkpoint state`.

Reply:
Added